exo/main.py

```diff
@@ -133,17 +134,21 @@
   lambda req_id, tokens, __: topology_viz.update_prompt_output(req_id, inference_engine.tokenizer.decode(tokens)) if topology_viz and hasattr(inference_engine, "tokenizer") else None
 )
 
-def preemptively_start_download(request_id: str, opaque_status: str):
+
+async def preemptively_start_download(request_id: str, opaque_status: str):
```

Comment:
I don't think this should be made async.
creating tasks fire-and-forget-style like it was before is better. if you need to run stuff in sequence, create a task for an asynchronous function that awaits each step.